src/ast/ast_validations.c

```diff
@@ -1659,17 +1655,19 @@ static AST_Validation _AST_ValidateUnionQuery(AST *mock_ast, char **reason) {
 	return res;
 }
 
-AST_Validation AST_Validate_Query(RedisModuleCtx *ctx, const cypher_parse_result_t *result) {
-	if(_AST_Validate_ParseResultRoot(ctx, result) != AST_VALID) return AST_INVALID;
+AST_Validation AST_Validate_Query(const cypher_parse_result_t *result) {
+	char *err = NULL;
+	if(_AST_Validate_ParseResultRoot(result, &err) != AST_VALID) {
+		QueryCtx_SetError(err);
```

Comment:
Wouldn't you prefer not to pass `err` and call `QueryCtx_SetError` wherever the error is detected?

Reply:
I agree with this, we have `QueryCtx_SetError` so that we don't need to store these `char **`s - we should write errors into the context upon encountering them.

Other than that, this PR looks good to me!